packages/components/src/Breakpoint/Breakpoint.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2020 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import React, { useContext, FC, ReactNode, useState } from 'react'
+import {
+  convertRemToPx,
+  NamedBreakpoints,
+  NAMED_BREAKPOINTS,
+} from '@looker/design-tokens'
+import isArray from 'lodash/isArray'
+import { ThemeContext } from 'styled-components'
+import { useResize } from '../utils'
+
+export interface BreakpointProps {
+  children: ReactNode
+  show: NamedBreakpoints | [NamedBreakpoints, NamedBreakpoints]
+}
+
+export const Breakpoint: FC<BreakpointProps> = ({ children, show }) => {
+  // Define screen size range.
+  // If they pass a single value, e.g. 'mobile', it should be equivalent to
+  // "from mobile, to mobile"
+  const [from = 'mobile', to = 'xl'] = isArray(show) ? show : [show, show]
+  const [screenWidth, setScreenWidth] = useState(document.body.clientWidth)
+  const theme = useContext(ThemeContext)
+  const breakpointPx = theme.breakpoints.map((b: string) =>
+    convertRemToPx(parseInt(b.replace('rem', '')))
+  )
+  const fromIndex = NAMED_BREAKPOINTS.indexOf(from)
+  const toIndex = NAMED_BREAKPOINTS.indexOf(to)
+
+  const handleResize = () => {
+    setScreenWidth(document.body.clientWidth)
+  }
+
+  useResize(document.body, handleResize)
+
+  const screenMin = from === 'mobile' ? 0 : breakpointPx[fromIndex - 1] // mobile screens start at 0px
+  const screenMax = to === 'xl' ? Infinity : breakpointPx[toIndex] // xl includes xl breakpoint and above
+
+  return (
+    <>{screenWidth > screenMin && screenWidth <= screenMax ? children : null}</>
+  )
```

Comment:
Does this need to be wrapped in a fragment or would this work?

```suggestion
  return screenWidth > screenMin && screenWidth <= screenMax ? children : null
```

Reply:
It depends. Looks like the FC type sets `children` as optional. So if I remove children as a required prop and remove the fragment, it complains that this might return `undefined` which is not a valid react node. 

I think it makes sense to enforce children as required, so I think I'll remove the fragment wrapper but leave the required `children: ReactNode` prop on line 38

Reply:
oh wait, nevermind. the `ReactNode` definition itself allows for `undefined` 🤦 

I'll remove line 38, but leave the fragment wrapper here. Here's the typescript error I get when I remove the wrapper:

```
const Breakpoint: React.FC<BreakpointProps>
Type '({ children, show }: PropsWithChildren<BreakpointProps>) => string | number | ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>)> | null' is not assignable to type 'FC<BreakpointProps>'.
  Type 'string | number | ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>)> | null' is not assignable to type 'ReactElement<any, any> | null'.
    Type 'string' is not assignable to type 'ReactElement<any, any> | null'.
```